### PR TITLE
Make better use of the rtt-target feature

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
- The rtt-target feature did not by itself initialize rtt, as this was configured by log-rtt
- Use rtt-target::rprint for panic and hardfaults that way the errors can still be obtained without going through delog

### CODE REVIEW

**Review by robin-nitrokey — APPROVED** [review]

**Review by daringer — APPROVED** [review]

**Comment by sosthene-nitrokey** [comment]
Huh, merge conflicts required a lot of changes.

File: runners/embedded/src/lib.rs [hunk 1]
```diff
@@ -207,9 +208,20 @@ pub fn init_se050<
...
     loop {
         core::sync::atomic::compiler_fence(core::sync::atomic::Ordering::SeqCst);
     }
 }
+
+unsafe fn handle_hard_fault<B: Board>(_ef: &ExceptionFrame) -> ! {
+    #[cfg(feature = "rtt-target")]
+    rtt_target::rprint!("HardFault: {:?}", _ef);
+    B::Led::set_panic_led();
+    soc::board::set_panic_led();
```
Comment (robin-nitrokey) [inline runners/embedded/src/lib.rs:223]: outdated line?

File: runners/embedded/src/bin/app-lpc.rs [hunk 2]
```diff
@@ -340,3 +342,8 @@ mod app {
 fn panic(info: &core::panic::PanicInfo) -> ! {
     embedded_runner_lib::handle_panic::<boards::nk3xn::NK3xN>(info)
 }
+
+#[exception]
+unsafe fn HardFault(ef: &ExceptionFrame) -> ! {
```
Comment (robin-nitrokey) [inline runners/embedded/src/bin/app-lpc.rs:347]: missing `ExceptionFrame` import?

File: runners/embedded/src/bin/app-nrf.rs [hunk 3]
```diff
@@ -336,3 +338,8 @@ mod app {
 fn panic(info: &core::panic::PanicInfo) -> ! {
     embedded_runner_lib::handle_panic::<boards::nk3am::NK3AM>(info)
 }
+
+#[exception]
+unsafe fn HardFault(ef: &ExceptionFrame) -> ! {
```
Comment (robin-nitrokey) [inline runners/embedded/src/bin/app-nrf.rs:343]: missing `ExceptionFrame` import?

**Review by robin-nitrokey — APPROVED** [review]

**Pull request merged**